code/modules/mob/mob.dm

```diff
@@ -49,6 +49,11 @@
 		mind.set_current(null)
 	return ..()
 
+/mob/New()
+	// This needs to happen IMMEDIATELY. I'm sorry :(
+	GenerateTag()
+	return ..()
+
```

Comment:
not changing request. I feel there should be something than this appraoch if GenerateTag should happen always first...